src/common/config.cc

```diff
@@ -473,6 +473,31 @@ void md_config_t::parse_env(unsigned entity_type,
       _set_val(values, tracker, dir, *o, CONF_ENV, &err);
     }
   }
+
+  // Apply pod memory limits:
+  //
+  // There are two types of resource requests: `limits` and `requests`.
+  //
+  // - Requests: Used by the K8s scheduler to determine on which nodes to
+  //   schedule the pods. This helps spread the pods to different nodes. This
+  //   value should be conservative in order to make sure all the pods are
+  //   schedulable.
+  // - Limits: At runtime, K8s will use the limits to see if the pod is using
```

Comment:
@travisn isn't it Linux that's monitoring the pod's containers and not K8s?

Reply:
Good clarification, it's the container runtime enforcing this, not K8s. Here's a more descriptive link for how this is enforced: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#how-pods-with-resource-limits-are-run